drawers/cMouseDrawer.cpp

```diff
@@ -190,58 +190,58 @@ void cMouseDrawer::drawToolTipGeneralInformation(cAbstractStructure * theStructu
 	} else {
 		sprintf(description, "%s", structureType.name);
 	}
-	textWriter->write(description, makecol(255, 255, 0));
-	textWriter->writeWithTwoIntegers("Hitpoints : %d/%d", theStructure->getHitPoints(), theStructure->getMaxHP());
-	textWriter->writeWithOneInteger("Armor : %d", theStructure->getArmor());
-	textWriter->writeWithOneInteger("Protected : %d%%", (100-theStructure->getPercentageNotPaved()));
+	textWriter.write(description, makecol(255, 255, 0));
+	textWriter.writeWithTwoIntegers("Hitpoints : %d/%d", theStructure->getHitPoints(), theStructure->getMaxHP());
+	textWriter.writeWithOneInteger("Armor : %d", theStructure->getArmor());
+	textWriter.writeWithOneInteger("Protected : %d%%", (100-theStructure->getPercentageNotPaved()));
 }
 
-void cMouseDrawer::drawToolTipWindTrapInformation(cWindTrap * theWindTrap, cTextWriter *textWriter) {
+void cMouseDrawer::drawToolTipWindTrapInformation(cWindTrap * theWindTrap, cTextWriter& textWriter) {
 	assert(theWindTrap);
-	assert(textWriter);
 	if (theWindTrap->getOwner() == HUMAN) {
 		int powerOut = theWindTrap->getPlayer()->getPowerProduced();
 		int powerUse = theWindTrap->getPlayer()->getPowerUsage();
 
 		if (powerUse <= powerOut) {
-			textWriter->writeWithTwoIntegers("Total usage : %d/%d (OK)", powerUse, powerOut);
+			textWriter.writeWithTwoIntegers("Total usage : %d/%d (OK)", powerUse, powerOut);
 		} else {
-			textWriter->writeWithTwoIntegers("Total usage : %d/%d (LOW)", powerUse, powerOut);
+			textWriter.writeWithTwoIntegers("Total usage : %d/%d (LOW)", powerUse, powerOut);
 		}
-		textWriter->writeWithTwoIntegers("Windtrap outage : %d/%d", theWindTrap->getPowerOut(), theWindTrap->getMaxPowerOut());
+		textWriter.writeWithTwoIntegers("Windtrap outage : %d/%d", theWindTrap->getPowerOut(), theWindTrap->getMaxPowerOut());
 	} else {
-		textWriter->write("Total usage : Unknown");
-		textWriter->write("Windtrap outage : Unknown");
+		textWriter.write("Total usage : Unknown");
+		textWriter.write("Windtrap outage : Unknown");
 	}
 }
 
-void cMouseDrawer::drawToolTipSiloInformation(cAbstractStructure * theStructure, cTextWriter *textWriter) {
+void cMouseDrawer::drawToolTipSiloInformation(cAbstractStructure * theStructure, cTextWriter& textWriter) {
 	assert(theStructure);
-	assert(textWriter);
-
+	
 	// TODO / IDEA --> Perhaps some 'spy on enemy intel' upgrade should be available for
 	// house Ordos, so you can actually check on the enemy spice etc.
 	if (theStructure->getOwner() == HUMAN) {
 		int spiceCapacityOfStructure = 0;
 		if (theStructure->getType() == REFINERY) {
 			cRefinery * refinery = dynamic_cast<cRefinery*>(theStructure);
+            assert(refinery);
```

Comment:
Additional asserts added? 

I do wonder if we can't solve this prettier. Instead of relying on `type` (REFINERY for instance), we should perhaps have some other way to get this information. Not sure how. My gut feel tells me that a sort of `composit` would make it possible to add some behavior to some structures which have a `capacity` of some sort. But I didn't spend too many brain cycles on this. Do you have any ideas about this?

Reply:
Yeah I can imagine. It feels awkward because we only have 1 or 2 buildings which will actually implement this :/